### ISSUE TITLE
SPSearchResultSource

### ISSUE THREAD

**Issue by NikCharlebois** [PowerShell/SharePointDsc#869]
#### Details of the scenario you tried and the problem that is occurring
It is possible to define Search Results at the Web level. Those are still being stored in the MSSSource table of the Search Service Application DB, so I wouldn't consider this to be content, but config. The current ressource only deals with Result Sources that belong to the SSA level Owner. To me, a combination of DSC and Content DB restore should allow anyone to replicate a Farm at 100% fidelity. In this case, web level Result Source fall into a grey zone, where since they are not contained within the Content DB, they would need to rely on DSC to be recreated, but DSC doesn't support them at the web level.

#### Verbose logs showing the problem
N/A

#### Suggested solution to the issue
The ressource should allow us to specify a scope: SSA or SPWeb. If SPWeb, then an optional URL parameter should be specified. I have developed the following code in ReverseDSC to retrieve Web level Result Sources. We could leverage it in our solution.

```powershell
$fedman = New-Object Microsoft.Office.Server.Search.Administration.Query.FederationManager($ssa)
$searchOwner = Get-SPEnterpriseSearchOwner -Level SPWeb -SPWeb $web
$filter = New-Object Microsoft.Office.Server.Search.Administration.SearchObjectFilter($searchOwner)
$filter.IncludeHigherLevel = $false
$sources = $fedman.ListSources($filter,$true)
```
#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
N/A

**Comment by NikCharlebois** [comment]
@ykuijs we need to brainstorm on how to better approach this. I believe this will have to be a breaking change no matter what. My ultimate goal here is that with a mix ReverseDSC/Forward DSC + Content DB copy, users can replicate their On-premises farm in Azure with full confidence.

**Comment by NikCharlebois** [comment]
Yes, agreed. Since all we can currently do is define SSA level items, setting it as the default value should work just fine. Good call.

**Comment by ykuijs** [comment]
Closed via PR #909
